[2/3] 3D Composability - move pp tests

mori360 opened the pull request:
pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
**PP:
test/distributed/pipelining/test_composability.py**

=>
distributed/_composable/test_composability/test_2d_composability.py
**distributed/_composable/test_composability/test_pp_composability.py**

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #129802
* __->__ #129801


cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

mori360 committed:
Update

[ghstack-poisoned]

kwen2501 commented:
why the move?

mori360 commented:
> why the move?

Thank you for the comments. The PR here is the start task of: Enable torchtitan jobs in PyTorch CI to catch PT-D breakage.
That we first collect the PyTorch tests together in a compatibility folder under PyTorch and then add the torchtitan jobs inside that folder.

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

wconstab commented on test/distributed/_composable/test_composability/test_pp_composability.py:
```diff
@@ -1,12 +1,9 @@
-# Copyright (c) Meta Platforms, Inc. and affiliates
```
the folder name seems too long to me.  why have test/distributed/_composable/test_composability/*?

why not just 

test/distributed/composability/* ?

Not sure if we have a strict rule that test names must match python module names.  but i would be ok ignoring that rule and just keep it simple.

wconstab commented on .ci/pytorch/multigpu-test.sh:
```diff
@@ -51,11 +51,9 @@ time python test/run_test.py --verbose -i distributed/tensor/parallel/test_tp_ra
-
 # 3D composability tests
 time python test/run_test.py --verbose -i distributed/_composable/test_composability/test_2d_composable
+time python test/run_test.py --verbose -i distributed/_composable/test_composability/test_pp_composable
```
a nit on file naming

it is not really clear which parallelisms should go into 2d and why pp is separate.

for now i think the pp ones are separate mostly because they are using the continuous test base and it doesn't mix nicely with older tests in the same file.

do you have another reason to separate files or is it just becuase of the issue i mentioned?

mori360 replied:
Thank you for the comments. 
That's the only reason for the separating right now.

wconstab commented:
> why the move?

I think we discussed at one point that it's not clear where to put the tests that focus on composing multiple things together, and as a result its also not clear what combinations we have tests for.  For example, when we do PP+FSDP tests, we might put them in the PP folder becuase we are thinking about PP at the time.  But it seems worthwhile to have a dedicated home for various composability tests, and then we focus more on 'unit' tests in the individual component folders.  Does it sound ok or do you have concerns?  (@kwen2501)

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update

[ghstack-poisoned]

mori360 committed:
Update on "[2/3] 3D Composability - move pp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
**PP:
test/distributed/pipelining/test_composability.py**

=>
distributed/_composable/test_composability/test_2d_composability.py
**distributed/_composable/test_composability/test_pp_composability.py**




cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

mori360 committed:
Update on "[2/3] 3D Composability - move pp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
**PP:
test/distributed/pipelining/test_composability.py**

=>
distributed/_composable/test_composability/test_2d_composability.py
**distributed/_composable/test_composability/test_pp_composability.py**




cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

wconstab reviewed (approved):
pp move lgtm!

mori360 committed:
Update on "[2/3] 3D Composability - move pp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
**PP:
test/distributed/pipelining/test_composability.py**

=>
distributed/_composable/test_composability/test_2d_composability.py
**distributed/_composable/test_composability/test_pp_composability.py**




cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

mori360 committed:
Update on "[2/3] 3D Composability - move pp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
**PP:
test/distributed/pipelining/test_composability.py**

=>
distributed/_composable/test_composability/test_2d_composability.py
**distributed/_composable/test_composability/test_pp_composability.py**




cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

mori360 committed:
Update on "[2/3] 3D Composability - move pp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
**PP:
test/distributed/pipelining/test_composability.py**

=>
distributed/_composable/test_composability/test_2d_composability.py
**distributed/_composable/test_composability/test_pp_composability.py**




cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

mori360 committed:
Update on "[2/3] 3D Composability - move pp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
**PP:
test/distributed/pipelining/test_composability.py**

=>
distributed/_composable/test_composability/test_2d_composability.py
**distributed/_composable/test_composability/test_pp_composability.py**




cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

mori360 committed:
Update on "[2/3] 3D Composability - move pp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
**PP:
test/distributed/pipelining/test_composability.py**

=>
distributed/_composable/test_composability/test_2d_composability.py
**distributed/_composable/test_composability/test_pp_composability.py**




cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

mori360 commented:
@pytorchbot merge

mori360 commented:
@pytorchbot merge

mori360 commented:
@pytorchbot merge

mori360 commented:
@pytorchbot merge

atalman commented:
@pytorchmergebot revert -c nosignal -m "Broke periodic CI: distributed/_composable/test_composability/test_pp_composability.py::ComposabilityTest::test_manual_with_data_parallel_dp_type_DDP_ScheduleClass4 [GH job link](https://github.com/pytorch/pytorch/actions/runs/10083807511/job/27882848654) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/544f950d14e8ce4da0e592d9a3faac349e718151)"

atalman commented:
@mori360  this can be relanded. PR that broke periodic CI is: https://github.com/pytorch/pytorch/pull/130752 sorry for reverting ths.

mori360 committed:
Update on "[2/3] 3D Composability - move pp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
**PP:
test/distributed/pipelining/test_composability.py**

=>
distributed/_composable/test_composability/test_2d_composability.py
**distributed/_composable/test_composability/test_pp_composability.py**




cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

mori360 committed:
Update on "[2/3] 3D Composability - move pp tests"

pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
DP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
TP:
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
**PP:
test/distributed/pipelining/test_composability.py**

=>
distributed/_composable/test_composability/test_2d_composability.py
**distributed/_composable/test_composability/test_pp_composability.py**



cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

mori360 commented:
@pytorchbot merge
